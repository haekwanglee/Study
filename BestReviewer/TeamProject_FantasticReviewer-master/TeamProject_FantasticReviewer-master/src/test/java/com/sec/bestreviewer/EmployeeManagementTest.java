package com.sec.bestreviewer;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;


import java.io.File;
import java.io.FileOutputStream;
import java.io.OutputStream;
import java.text.Format;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class EmployeeManagementTest {
    @Test
    public void testArgumentsEmptyInputFile() throws Exception {
        String[] args = {"./src/test/java/com/sec/bestreviewer/empty.txt", "./src/test/java/com/sec/bestreviewer/output.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();
        employeeManagement.run(args);

        File file = new File("./src/test/java/com/sec/bestreviewer/output.txt");

        assertTrue(file.exists());
    }

    @Test
    public void testArgumentsWrongArgsCount() throws Exception {
        String[] args = {"input.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();

        Assertions.assertThrows(IllegalArgumentException.class, () -> {
            employeeManagement.run(args);
        });
    }

    @Test
    public void testArgumentsNotExistInputFile() throws Exception {
        String[] args = {"notexist.txt", "output.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();

        Assertions.assertThrows(IllegalArgumentException.class, () -> {
            employeeManagement.run(args);
        });
    }

    @Test
    public void testMainArgumentsNotExistInputFile() throws Exception {
        String[] args = {"notexist.txt", "output.txt"};

        Assertions.assertThrows(IllegalArgumentException.class, () -> {
            EmployeeManagement.main(args);
        });
    }

    @Test
    public void integrationVerySmallDataTest() throws Exception {
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/big-data/very-small_output.txt";
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/big-data/very-small-data.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    public void integrationSmallDataTest() throws Exception {
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/big-data/small_output.txt";
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/big-data/small-data.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    public void integrationMiddleDataTest() throws Exception {
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/big-data/middle_output.txt";
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/big-data/middle-data.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    public void integrationLargeDataTest() throws Exception {
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/big-data/large_output.txt";
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/big-data/large-data.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

}
