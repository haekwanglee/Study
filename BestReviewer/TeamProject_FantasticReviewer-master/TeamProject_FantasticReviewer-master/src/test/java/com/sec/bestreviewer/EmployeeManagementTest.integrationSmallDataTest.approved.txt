MOD,6
DEL,11
SCH,NONE
SCH,NONE
SCH,NONE
DEL,NONE
SCH,22
DEL,5
SCH,NONE
DEL,NONE
MOD,NONE
SCH,NONE
SCH,2
DEL,NONE
SCH,91761534,AZNQCVW KIM,CL1,010-4619-6000,19700825,PRO
SCH,93728440,TRLK KIM,CL1,010-1481-5313,19701111,PRO
SCH,94470950,XJO KIM,CL1,010-5474-1182,19670807,ADV
SCH,97754169,FKPBNM KIM,CL1,010-9534-4739,19740515,ADV
SCH,00072559,NUP KANG,CL3,010-6680-7743,19761013,ADV
DEL,NONE
MOD,NONE
SCH,NONE
SCH,24
DEL,NONE
SCH,10
SCH,NONE
SCH,91761534,AZNQCVW KIM,CL1,010-4619-6000,19700825,PRO
SCH,93728440,TRLK KIM,CL1,010-1481-5313,19701111,PRO
SCH,94470950,XJO KIM,CL1,010-5474-1182,19670807,ADV
SCH,97754169,FKPBNM KIM,CL1,010-9534-4739,19740515,ADV
SCH,15401757,YBVQO PARK,CL1,010-2012-5131,19920102,ADV
SCH,91761534,AZNQCVW KIM,CL1,010-4619-6000,19700825,PRO
SCH,93728440,TRLK KIM,CL1,010-1481-5313,19701111,PRO
SCH,94470950,XJO KIM,CL1,010-5474-1182,19670807,ADV
SCH,97754169,FKPBNM KIM,CL1,010-9534-4739,19740515,ADV
SCH,15401757,YBVQO PARK,CL1,010-2012-5131,19920102,ADV
SCH,91761534,AZNQCVW KIM,CL1,010-4619-6000,19700825,PRO
SCH,93728440,TRLK KIM,CL1,010-1481-5313,19701111,PRO
SCH,03298170,IHQM LEE,CL3,010-1242-3260,19820910,PRO
SCH,04818446,NFSBKIUSE CHO,CL3,010-1489-7041,19760307,PRO
SCH,05822143,IAF SEO,CL3,010-5951-5857,19771126,PRO
SCH,24
SCH,94470950,XJO KIM,CL1,010-5474-1182,19670807,ADV
SCH,97754169,FKPBNM KIM,CL1,010-9534-4739,19740515,ADV
SCH,00072559,NUP KANG,CL3,010-6680-7743,19761013,ADV
SCH,02755375,RKDQTS JANG,CL3,010-7593-3764,19740725,EX
SCH,03298170,IHQM LEE,CL3,010-1242-3260,19820910,PRO
SCH,NONE
SCH,NONE
SCH,97754169,FKPBNM KIM,CL1,010-9534-4739,19740515,ADV
SCH,08926021,BTJWPB CHOI,CL3,010-2449-4384,19850807,PRO
SCH,15653316,MLG KIM,CL1,010-3936-9889,19870915,ADV
MOD,1
SCH,14
MOD,NONE
DEL,6
SCH,NONE
SCH,8
SCH,NONE
DEL,NONE
MOD,NONE
MOD,NONE
